src/dbuffer.hpp

```diff
@@ -78,12 +78,12 @@ template <> class dbuffer_t<msg_t>
         msg_t &xvalue = const_cast<msg_t &> (value_);
 
         zmq_assert (xvalue.check ());
-        _back->move (xvalue); // cannot just overwrite, might leak
+        *_back = value_;
 
         zmq_assert (_back->check ());
 
         if (_sync.try_lock ()) {
-            std::swap (_back, _front);
+            _front->move (*_back);
```

Comment:
The commit message says:

> swap front and back buffer using msg_t::move instead of std::swap

But this does not swap them anymore, instead _back is de initialized. Is that intended?

Reply:
>But this does not swap them anymore, instead _back is de initialized. Is that intended?
Yes, the idea is to release the message in the front buffer first before moving the message in the back buffer into it. I just changed the commit message to reflect that intent better.